tests/components/apcupsd/__init__.py

```diff
@@ -35,6 +36,7 @@
         ("HITRANS", "139.0 Volts"),
         ("ALARMDEL", "30 Seconds"),
         ("BATTV", "13.7 Volts"),
+        ("OUTCURNT", "0.88 Amps"),
```

Comment:
"New tests for these two units are added to prevent future regression."

Are these the new tests?🤔

Reply:
Yes, this tests if our unit inference can properly split the `Amps` from `0.88 Amps` (and subsequently convert `Amps` to `UnitOfElectricCurrent.AMPERE` in HA). Without the fix in this PR this would result in the following error:

```
ERROR:homeassistant.components.sensor:Error adding entities for domain sensor with platform apcupsd
Traceback (most recent call last):
  File "/Users/ryan/Desktop/core/homeassistant/components/sensor/__init__.py", line 583, in state
    numerical_value = float(value)  # type:ignore[arg-type]
ValueError: could not convert string to float: '0.88 Amps'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/ryan/Desktop/core/homeassistant/helpers/entity_platform.py", line 471, in async_add_entities
    await asyncio.gather(*tasks)
  File "/Users/ryan/Desktop/core/homeassistant/helpers/entity_platform.py", line 749, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/Users/ryan/Desktop/core/homeassistant/helpers/entity.py", line 869, in add_to_platform_finish
    self.async_write_ha_state()
  File "/Users/ryan/Desktop/core/homeassistant/helpers/entity.py", line 610, in async_write_ha_state
    self._async_write_ha_state()
  File "/Users/ryan/Desktop/core/homeassistant/helpers/entity.py", line 673, in _async_write_ha_state
    state = self._stringify_state(available)
  File "/Users/ryan/Desktop/core/homeassistant/helpers/entity.py", line 616, in _stringify_state
    if (state := self.state) is None:
  File "/Users/ryan/Desktop/core/homeassistant/components/sensor/__init__.py", line 585, in state
    raise ValueError(
ValueError: Sensor sensor.ups_output_current has device class 'current', state class 'measurement' unit 'A' and suggested precision 'None' thus indicating it has a numeric value; however, it has the non-numeric value: '0.88 Amps' (<class 'str'>)
```